\documentclass[12pt,t]{beamer}
\usepackage{graphicx}
\setbeameroption{hide notes}
\setbeamertemplate{note page}[plain]
\usepackage{listings}
\usepackage{datetime}
\usepackage{url}

% specifications for presenter mode
%\beamerdefaultoverlayspecification{<+->}
%\setbeamercovered{transparent}

% math shorthand
\usepackage{bm}
\usepackage{amsmath}
\usepackage{mathtools}
\newcommand{\D}{\mathcal{D}}
\newcommand{\E}{\mathbb{E}}
\newcommand{\F}{\mathcal{F}}
\newcommand{\X}{\mathcal{X}}
\newcommand{\lik}{\mathcal{L}}
\DeclarePairedDelimiterX{\infdivx}[2]{(}{)}{%
  #1\;\delimsize\|\;#2%
}
\newcommand{\infdiv}{D\infdivx}
\DeclarePairedDelimiter{\norm}{\lVert}{\rVert}
\DeclareMathOperator*{\argmin}{arg\,min}
\DeclareMathOperator*{\argmax}{arg\,max}

% Bibliography
\usepackage{natbib}
\bibpunct{(}{)}{,}{a}{}{;}
\usepackage{bibentry}
\nobibliography*

\input{header.tex}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% end of header
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% title info
\title{\large Data-Adaptive Estimation and Inference in the Analysis of
  Differential Methylation}
\subtitle{\scriptsize for the annual retreat of the \textit{Center for
                      Computational Biology},\\
                    given 18 November, 2017
                    \\[-10pt]
         }
\author{\href{http://nimahejazi.org}{Nima Hejazi}
       \\[-10pt]
       }
\institute{Division of Biostatistics \\
           University of California, Berkeley \\
           \href{https://www.stat.berkeley.edu/~nhejazi}
             {\tt \scriptsize \color{foreground}
               stat.berkeley.edu/\textasciitilde{}nhejazi
             }
           \\[4pt]
           \includegraphics[height=20mm]{Figs/seal-berkeley.png}
           \\[-12pt]
          }
\date{
  \href{http://nimahejazi.org}
      {\tt \scriptsize \color{foreground} nimahejazi.org}
  \\[-4pt]
  \href{https://twitter.com/nshejazi}
      {\tt \scriptsize \color{foreground} twitter/@nshejazi}
  \\[-4pt]
  \href{https://github.com/nhejazi}
      {\tt \scriptsize \color{foreground} github/nhejazi}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}

% title slide
{
\setbeamertemplate{footline}{} % no page number here
\frame{
  \titlepage

  \vspace{-1em}

  \centerline{\href{https://goo.gl/xabp3Q}{\tt \scriptsize
                                           \underline{slides}: goo.gl/xabp3Q}}
  \vspace{-1.5em}
  \vfill \hfill \includegraphics[height=6mm]{Figs/cc-zero.png} \vspace*{-0.5cm}

  \note{This slide deck is for a brief (about 15-minute) talk on a new
    statistical algorithm for using nonparametric and data-adaptive estimates of
    variable importance measures for differential methylation analysis. This
    talk was most recently given at the annual retreat of the
    \href{http://ccb.berkeley.edu/}{Center for
    Computational Biology} at the University of California, Berkeley.

    Source: {\tt https://github.com/nhejazi/talk\_methyvim} \\
    Slides: {\tt https://goo.gl/JDhSEg} \\
    With notes: {\tt https://goo.gl/xabp3Q}
}
}
}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Preview: Summary}
\only<1>{\addtocounter{framenumber}{-1}}

\begin{center}
\begin{itemize}
  \itemsep12pt
  \item DNA methylation data is \textit{extremely} high-dimensional --- we can
    collect data on 850K genomic sites with modern arrays!
  \item Normalization is a critical component of properly analyzing modern DNA
    methylation data sets, and there are many choices of technique.
  \item There is a relative scarcity of techniques for estimation and inference
    --- analyses are often limited to the general linear model.
  \item Statistical causal inference provides an avenue for answering richer
    scientific questions, especially when combined with modern advances in
    machine learning.
\end{itemize}
\end{center}

\note{We'll go over this summary again at the end of the talk. Hopefully, it
  will all make more sense then.
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[fragile,c]{}

\begin{center}
\begin{minipage}[c]{9.3cm}
\begin{semiverbatim}
\lstset{basicstyle=\normalsize}
\begin{lstlisting}[linewidth=9.3cm]
  It's always good to start with a
  motivating example. Excerpts from
  famous studies/papers or personal
  communications usually do nicely.

  It's also good practice to keep
  things like this rather short.

  --Nima
\end{lstlisting}
\end{semiverbatim}
\end{minipage}
\end{center}

\note{Obviously, it's important to explain the motivating example here.}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Motivation: Let's meet the data}

\begin{center}
\begin{itemize}
  \itemsep12pt
  \item Observational study of the impact of occupational exposure to benzene on
    DNA methylation.
  \item Phenotype-level quantities: $216$ subjects, binary exposure status of
    each subject, background info on subjects (e.g., sex, age).
  \item Genomic-level quantities: $\sim 850,000$ CpG sites interrogated using
    the \textit{Infinium MethylationEPIC BeadChip} by Illumina.
  \item \textbf{Question}: How does exposure induce differential methylation? Is
    a coherent biomarker-type signature detectable?
\end{itemize}
\end{center}

\note{
...
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Data analysis? Not yet. Normalization!}

\begin{center}
\begin{itemize}
  \item Need to normalize the data
  \item Blah blah blah
  \item Images from Rachael
\end{itemize}
\end{center}

\note{
...
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Data analysis? It's hard.}
\begin{center}
\begin{itemize}
  \itemsep12pt
  \item SOP: For each biomarker ($b = 1, \dots, B$), fit a linear model:
    \[
    \mathbb{E}[y_b] = X \beta_b
    \]
  \item SOP: Test the coefficent of interest using a standard t-test:
    \[
      t_{b} = \frac{\hat{\beta}_{b} - \beta_{b, H_0}}{s_b}
    \]
  \item Such models are a matter of convenience: does $\hat{\beta}_{b}$ really
    answer our questions?
  \item Treating CpG sites as acting independently runs counter to biological
    notions of such sites functioning in networks.
\end{itemize}
\end{center}


\note{
}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Target parameters for complex questions}

\begin{center}
\begin{itemize}
  \itemsep12pt
  \item Rather than being satisfied with $\hat{\beta}_{b}$ as an answer to our
    questions, let's consider a simple target parameter: the average treatment
    effect (ATE):
    \[
      \Psi_b(P_0) = \mathbb{E}_{W,0}[\mathbb{E}_0[Y_b \mid A = a_{high}, W] -
      \mathbb{E}_0[Y_b \mid A = a_{low}, W]]
    \]
  \item No need to specify a functional form or assume that we know the true
    data-generating distribution $P_0$.
  \item Parameters like this can be estimated using \textit{targeted minimum
    loss-based estimation} (TMLE).
  \item \textbf{Asymptotic linearity:}
    \[
      \Psi_b(P_n^*) - \Psi_b(P_0) = \frac{1}{n} \sum_{i = 1}^{n} IC(O_i) +
      o_P(\frac{1}{\sqrt{n}})
    \]
\end{itemize}
\end{center}

\note{By allowing scientific questions to inform the parameters that we choose
      to estimate, we can do a better job of actually answering the questions of
      interest to our collaborators. Further, we abandon the need to specify the
      functional relationship between our outcome and covariates; moreover, we
      can now make use of advances in machine learning.
}
\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Data analysis? A data-adaptive approach.}

\begin{center}
\begin{enumerate}
  \item Pre-screening of genomic sites is used to isolate a subset of sites for
    which there is cursory evidence of differential methylation.
  \item Nonparametric estimates of VIMs, for the specified target parameter, are
   computed at each of the CpG sites passing the screening step. The VIMs are
   defined in such a way that the estimated effects is of an exposure/treatment
   on the methylation status of a target CpG site, controlling for the observed
   methylation status of the neighbors of that site.
 \item When there are too many neighbors, Partitioning Around Medoids (PAM) is
   used to select the largest number of neighbors that can be controlled for.
 \item Since pre-screening is performed prior to estimating VIMs, we make use of a
   multiple testing correction uniquely suited to such settings. Due to the
   multiple testing nature of the estimation problem, a variant of the Benjamini
   \& Hochberg procedure for controlling the False Discovery Rate (FDR) is
   applied
\end{enumerate}
\end{center}

\note{
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Pre-Screening}

\begin{center}
\begin{itemize}
  \item ...
  \item ...
\end{itemize}
\end{center}

\note{
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Too Many Neighbors? PAM}

\begin{center}
\begin{itemize}
  \item ...
  \item ...
\end{itemize}
\end{center}

\note{
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Nonparametric Estimation of a VIM}

\begin{center}
\begin{itemize}
  \item ...
  \item ...
\end{itemize}
\end{center}

\note{
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Correcting for Multiple Testing}

\begin{center}
\begin{itemize}
  \item ...
  \item ...
\end{itemize}
\end{center}

\note{
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Software package: R/``methyvim''}

\begin{center}
\begin{itemize}
  \item ...
  \item ...
\end{itemize}
\end{center}

\note{
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Data analysis the ``methyvim'' way}

\begin{center}
\begin{itemize}
  \item ...
  \item ...
\end{itemize}
\end{center}

\note{
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Data analysis results with ``methyvim'': p-values}

\begin{center}
\begin{itemize}
  \item ...
  \item ...
\end{itemize}
\end{center}

\note{
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Data analysis results with ``methyvim'': Volcano}

\begin{center}
\begin{itemize}
  \item ...
  \item ...
\end{itemize}
\end{center}

\note{
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Data analysis results with ``methyvim'': Heatmap}

\begin{center}
\begin{itemize}
  \item ...
  \item ...
\end{itemize}
\end{center}

\note{
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Review: Summary}

\begin{center}
\begin{itemize}
  \itemsep12pt
  \item Look, we proved this above.
  \item More stuff we proved.
  \item Yet another point here.
  \item Final point goes here.
\end{itemize}
\end{center}

\note{It's always good to include a summary.}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

% don't want dimming with references
\setbeamercovered{}
\beamerdefaultoverlayspecification{}

\begin{frame}[c,allowframebreaks]{References}

\bibliographystyle{apalike}
\nocite{*}
\bibliography{references}

%\note{Here's some work we've talked about. Go check these out if interested.}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}{Acknowledgments}

\vspace{18pt}

\begin{tabular}{@{}l@{\hspace{1.5cm}}l@{}}
Collaborator, the first & \footnotesize \lolit University or Institution 1 \\
Collaborator, the second \\
Collaborator, the third \\

\\[2ex]

Collaborator, the first & \footnotesize \lolit University or Institution 2 \\
Collaborator, the second \\
\end{tabular}

\vspace{10mm}

Funding source?

\note{
}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{frame}[c]{Thank you.}

\Large
Slides: \href{https://goo.gl/JDhSEg}{goo.gl/JDhSEg} \quad
\includegraphics[height=5mm]{Figs/cc-zero.png}

\vspace{3mm}
Notes: \href{https://goo.gl/xabp3Q}{goo.gl/xabp3Q}

\vspace{3mm}
Source (repo): \href{https://goo.gl/m5As73}{goo.gl/m5As73}

\vspace{3mm}
\href{https://www.stat.berkeley.edu/~nhejazi}{\tt
  stat.berkeley.edu/\textasciitilde{}nhejazi}

\vspace{3mm}
\href{http://nimahejazi.org}{\tt nimahejazi.org}

\vspace{3mm}
\href{https://twitter.com/nshejazi}{\tt twitter/@nshejazi}

\vspace{3mm}
\href{https://github.com/nhejazi}{\tt github/nhejazi}

\note{Here's where you can find me, as well as the slides for this talk.}

\end{frame}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\end{document}

